samza-azure/src/test/java/org/apache/samza/system/azureblob/avro/TestAzureBlobAvroWriter.java

```diff
@@ -0,0 +1,525 @@
+
+    OutgoingMessageEnvelope ome2 = createOME("Topic2");
+    DataFileWriter mockDataFileWriter2 = mock(DataFileWriter.class);
+    PowerMockito.whenNew(DataFileWriter.class).withAnyArguments().thenReturn(mockDataFileWriter2);
+
+    BlobAsyncClient mockBlobAsyncClient2 = mock(BlobAsyncClient.class);
+    doReturn(mockBlobAsyncClient2).when(mockContainerClient).getBlobAsyncClient(anyString());
+    BlockBlobAsyncClient mockBlockBlobAsyncClient2 = mock(BlockBlobAsyncClient.class);
+    doReturn(mockBlockBlobAsyncClient2).when(mockBlobAsyncClient2).getBlockBlobAsyncClient();
+
+    AzureBlobOutputStream mockAzureBlobOutputStream2 = mock(AzureBlobOutputStream.class);
+    PowerMockito.whenNew(AzureBlobOutputStream.class).withArguments(mockBlockBlobAsyncClient2, threadPool,
+        mockMetrics,
+        (long) 60000, THRESHOLD, mockCompression).thenReturn(mockAzureBlobOutputStream2);
+    when(mockAzureBlobOutputStream2.getSize()).thenReturn((long) maxBlobSize - 1);
+
+    // Second OME creates the second blob because maxBlobSize is 1000 and mockAzureBlobOutputStream.getSize is 999.
+    azureBlobAvroWriter.write(ome2);
+
+    ArgumentCaptor<String> argument = ArgumentCaptor.forClass(String.class);
+    verify(mockContainerClient, times(2)).getBlobAsyncClient(argument.capture());
+    argument.getAllValues().forEach(blobName -> {
+        Assert.assertTrue(blobName.contains(blobUrlPrefix));
+      });
+    List<String> allBlobNames = argument.getAllValues();
+    Assert.assertNotEquals(allBlobNames.get(0), allBlobNames.get(1));
+
+    verify(mockDataFileWriter1).appendEncoded(ByteBuffer.wrap(encodeRecord((IndexedRecord) ome.getMessage())));
+    verify(mockDataFileWriter2).appendEncoded(ByteBuffer.wrap(encodeRecord((IndexedRecord) ome2.getMessage())));
+
+    verify(mockDataFileWriter1).create(((IndexedRecord) ome.getMessage()).getSchema(), mockAzureBlobOutputStream1);
+    verify(mockDataFileWriter2).create(((IndexedRecord) ome2.getMessage()).getSchema(), mockAzureBlobOutputStream2);
+  }
+
+  @Test
+  public void testRecordLimitExceeded() throws Exception {
+    String blobUrlPrefix = "test";
+    AzureBlobWriterMetrics mockMetrics = mock(AzureBlobWriterMetrics.class);
+    long maxBlobSize = AzureBlobAvroWriter.DATAFILEWRITER_OVERHEAD + 1000;
+    long maxRecordsPerBlob = 10;
+    BlobContainerAsyncClient mockContainerClient = PowerMockito.mock(BlobContainerAsyncClient.class);
+    azureBlobAvroWriter = spy(new AzureBlobAvroWriter(mockContainerClient,
+        mockMetrics, threadPool, THRESHOLD, 60000, blobUrlPrefix,
+        null, null, null, maxBlobSize, maxRecordsPerBlob, mockCompression, true));
+
+    DataFileWriter mockDataFileWriter1 = mock(DataFileWriter.class);
+    PowerMockito.whenNew(DataFileWriter.class).withAnyArguments().thenReturn(mockDataFileWriter1);
+
+    BlobAsyncClient mockBlobAsyncClient1 = mock(BlobAsyncClient.class);
+    doReturn(mockBlobAsyncClient1).when(mockContainerClient).getBlobAsyncClient(anyString());
```

Comment:
Would it be possible to check at least some part of the blob URL that you use to create the blob async client? For example, it should match some regex something like `test/[0-9]+/[0-9]+/[0-9]+/[0-9]+/[0-9]+-[0-9]+-.*.avro.gz`. This could verify some of the expected format and that there is some random string in there.